rust/lance/src/dataset/schema_evolution.rs

```diff
@@ -149,6 +155,16 @@ pub(super) async fn add_columns_to_fragments(
         Ok(())
     };
 
+    // ALlNull transform can not performed on legacy files
+    let has_legacy_files = has_legacy_files(fragments);
+
+    // Optimize the transforms
+    let mut optimizer = ChainedNewColumnTransformOptimizer::new(vec![]);
+    if has_legacy_files {
```

Comment:
Should this be `!has_legacy_files`?  Or am I misunderstanding the above comment?